crates/bevy_math/src/bounding/bounded3d/primitive_impls.rs

```diff
@@ -0,0 +1,568 @@
+//! Contains [`Bounded3d`] implementations for [geometric primitives](crate::primitives).
+
+use glam::{Mat3, Quat, Vec2, Vec3};
+
+use crate::{
+    bounding::{Bounded2d, BoundingCircle},
+    primitives::{
+        BoxedPolyline3d, Capsule, Cone, ConicalFrustum, Cuboid, Cylinder, Direction3d, Line3d,
+        Plane3d, Polyline3d, Segment3d, Sphere, Torus, Triangle2d,
+    },
+};
+
+use super::{Aabb3d, Bounded3d, BoundingSphere};
+
+impl Bounded3d for Sphere {
+    fn aabb_3d(&self, translation: Vec3, _rotation: Quat) -> Aabb3d {
+        Aabb3d {
+            min: translation - Vec3::splat(self.radius),
+            max: translation + Vec3::splat(self.radius),
+        }
+    }
+
+    fn bounding_sphere(&self, translation: Vec3, _rotation: Quat) -> BoundingSphere {
+        BoundingSphere::new(translation, self.radius)
+    }
+}
+
+impl Bounded3d for Plane3d {
+    fn aabb_3d(&self, translation: Vec3, rotation: Quat) -> Aabb3d {
+        let normal = rotation * *self.normal;
+        let facing_x = normal == Vec3::X || normal == Vec3::NEG_X;
+        let facing_y = normal == Vec3::Y || normal == Vec3::NEG_Y;
+        let facing_z = normal == Vec3::Z || normal == Vec3::NEG_Z;
+
+        // Dividing `f32::MAX` by 2.0 can actually be good so that we can do operations
+        // like growing or shrinking the AABB without breaking things.
+        let half_width = if facing_x { 0.0 } else { f32::MAX / 2.0 };
+        let half_height = if facing_y { 0.0 } else { f32::MAX / 2.0 };
+        let half_depth = if facing_z { 0.0 } else { f32::MAX / 2.0 };
+        let half_size = Vec3::new(half_width, half_height, half_depth);
+
+        Aabb3d {
+            min: translation - half_size,
+            max: translation + half_size,
+        }
+    }
+
+    fn bounding_sphere(&self, translation: Vec3, _rotation: Quat) -> BoundingSphere {
+        BoundingSphere::new(translation, f32::MAX / 2.0)
+    }
+}
+
+impl Bounded3d for Line3d {
+    fn aabb_3d(&self, translation: Vec3, rotation: Quat) -> Aabb3d {
+        let direction = rotation * *self.direction;
+
+        // Dividing `f32::MAX` by 2.0 can actually be good so that we can do operations
+        // like growing or shrinking the AABB without breaking things.
+        let max = f32::MAX / 2.0;
+        let half_width = if direction.x == 0.0 { 0.0 } else { max };
+        let half_height = if direction.y == 0.0 { 0.0 } else { max };
+        let half_depth = if direction.z == 0.0 { 0.0 } else { max };
```

Comment:
This is because of formatting. If I change it to that, it's formatted like:
```rust
        let half_width = if direction.x == 0.0 {
            0.0
        } else {
            f32::MAX / 2.0
        };
        let half_height = if direction.y == 0.0 {
            0.0
        } else {
            f32::MAX / 2.0
        };
        let half_depth = if direction.z == 0.0 {
            0.0
        } else {
            f32::MAX / 2.0
        };
```